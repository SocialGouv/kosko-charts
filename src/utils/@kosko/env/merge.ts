//

/* eslint-disable @typescript-eslint/no-unsafe-return */
/* eslint-disable @typescript-eslint/no-explicit-any */

// HACK(douglasduteil): wrap internal @kosko/env/dist/merge.js
// As the paths API is internal (not exported) in @kosko/env
// This is code will wrap and expose it for our needs.
// see https://github.com/tommy351/kosko/blob/%40kosko/env%401.0.1/packages/env/src/merge.ts#L12-L18
// since https://github.com/tommy351/kosko/commit/6353c8559994b826c8056ad03a35332e88b9134a

// https://stackoverflow.com/a/48769843
type UnionToIntersection<T> = (T extends any ? (k: T) => void : never) extends (
  k: infer I
) => void
  ? I
  : never;

export const merge: <T extends any[]>(
  ...data: T
) => UnionToIntersection<T[number]> =
  // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
  require("@kosko/env/dist/merge.js").merge;
