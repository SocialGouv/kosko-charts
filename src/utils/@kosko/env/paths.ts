//

import { join } from "path";

// HACK(douglasduteil): wrap internal @kosko/env/dist/paths.js
// As the paths API is internal (not exported) in @kosko/env
// This is code will wrap and expose it for our needs.
// see https://github.com/tommy351/kosko/blob/%40kosko/env%400.5.2/packages/env/src/paths.ts#L9-L13
// since https://github.com/tommy351/kosko/commit/6353c8559994b826c8056ad03a35332e88b9134a

export const formatPath: (
  path: string,
  data: Record<string, unknown>
) => string =
  // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
  require(join(require.resolve("@kosko/env"), "../paths.js")).formatPath;
