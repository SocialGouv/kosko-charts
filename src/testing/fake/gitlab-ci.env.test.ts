import { project } from "./gitlab-ci.env";

test("should generate foo gitlab project env", () => {
  const { base, dev, preprod, prod } = project("sample");

  expect(base).toMatchSnapshot("base");
  expect(dev).toMatchSnapshot("dev");
  expect(preprod).toMatchSnapshot("preprod");
  expect(prod).toMatchSnapshot("prod");
});
