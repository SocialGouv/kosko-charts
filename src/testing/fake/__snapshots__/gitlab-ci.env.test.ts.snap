// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate foo gitlab project env: base 1`] = `
Object {
  "CI_COMMIT_BEFORE_SHA": "0000000000000000000000000000000000000000",
  "CI_COMMIT_DESCRIPTION": "",
  "CI_COMMIT_MESSAGE": "chore(release): version 1.2.3",
  "CI_COMMIT_REF_PROTECTED": "false",
  "CI_COMMIT_SHA": "0123456789abcdefghijklmnopqrstuvwxyz0123",
  "CI_COMMIT_SHORT_SHA": "0123456",
  "CI_COMMIT_TIMESTAMP": "1111-11-11T11:11:11+00:00",
  "CI_COMMIT_TITLE": "chore(release): version 1.2.3",
  "CI_ENVIRONMENT_URL": "https://sample.fabrique.social.gouv.fr",
  "CI_PROJECT_DIR": "/builds/SocialGouv/sample",
  "CI_PROJECT_ID": "85",
  "CI_PROJECT_NAME": "sample",
  "CI_PROJECT_NAMESPACE": "SocialGouv",
  "CI_PROJECT_PATH": "SocialGouv/sample",
  "CI_PROJECT_PATH_SLUG": "socialgouv-sample",
  "CI_PROJECT_ROOT_NAMESPACE": "SocialGouv",
  "CI_PROJECT_TITLE": "sample",
  "CI_PROJECT_URL": "https://gitlab.factory.social.gouv.fr/SocialGouv/sample",
  "CI_PROJECT_VISIBILITY": "public",
  "CI_REGISTRY_IMAGE": "registry.gitlab.factory.social.gouv.fr/socialgouv/sample",
  "CI_REGISTRY_USER": "gitlab-ci-token",
  "CI_REPOSITORY_URL": "https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/sample.git",
}
`;

exports[`should generate foo gitlab project env: dev 1`] = `
Object {
  "CI_COMMIT_BEFORE_SHA": "0000000000000000000000000000000000000000",
  "CI_COMMIT_DESCRIPTION": "",
  "CI_COMMIT_MESSAGE": "chore(release): version 1.2.3",
  "CI_COMMIT_REF_NAME": "master",
  "CI_COMMIT_REF_PROTECTED": "false",
  "CI_COMMIT_REF_SLUG": "master",
  "CI_COMMIT_SHA": "0123456789abcdefghijklmnopqrstuvwxyz0123",
  "CI_COMMIT_SHORT_SHA": "0123456",
  "CI_COMMIT_TAG": "",
  "CI_COMMIT_TIMESTAMP": "1111-11-11T11:11:11+00:00",
  "CI_COMMIT_TITLE": "chore(release): version 1.2.3",
  "CI_ENVIRONMENT_NAME": "master-dev42",
  "CI_ENVIRONMENT_SLUG": "master-dev42",
  "CI_ENVIRONMENT_URL": "https://master-dev42-sample.dev42.fabrique.social.gouv.fr",
  "CI_PROJECT_DIR": "/builds/SocialGouv/sample",
  "CI_PROJECT_ID": "85",
  "CI_PROJECT_NAME": "sample",
  "CI_PROJECT_NAMESPACE": "SocialGouv",
  "CI_PROJECT_PATH": "SocialGouv/sample",
  "CI_PROJECT_PATH_SLUG": "socialgouv-sample",
  "CI_PROJECT_ROOT_NAMESPACE": "SocialGouv",
  "CI_PROJECT_TITLE": "sample",
  "CI_PROJECT_URL": "https://gitlab.factory.social.gouv.fr/SocialGouv/sample",
  "CI_PROJECT_VISIBILITY": "public",
  "CI_REGISTRY_IMAGE": "registry.gitlab.factory.social.gouv.fr/socialgouv/sample",
  "CI_REGISTRY_USER": "gitlab-ci-token",
  "CI_REPOSITORY_URL": "https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/sample.git",
  "KUBE_INGRESS_BASE_DOMAIN": "dev42.fabrique.social.gouv.fr",
  "KUBE_NAMESPACE": "sample-85-master-dev42",
  "PRODUCTION": "",
}
`;

exports[`should generate foo gitlab project env: preprod 1`] = `
Object {
  "CI_COMMIT_BEFORE_SHA": "0000000000000000000000000000000000000000",
  "CI_COMMIT_DESCRIPTION": "",
  "CI_COMMIT_MESSAGE": "chore(release): version 1.2.3",
  "CI_COMMIT_REF_NAME": "v1.2.3",
  "CI_COMMIT_REF_PROTECTED": "false",
  "CI_COMMIT_REF_SLUG": "v1-2-3",
  "CI_COMMIT_SHA": "0123456789abcdefghijklmnopqrstuvwxyz0123",
  "CI_COMMIT_SHORT_SHA": "0123456",
  "CI_COMMIT_TAG": "v1.2.3",
  "CI_COMMIT_TIMESTAMP": "1111-11-11T11:11:11+00:00",
  "CI_COMMIT_TITLE": "chore(release): version 1.2.3",
  "CI_ENVIRONMENT_NAME": "preprod-dev42",
  "CI_ENVIRONMENT_SLUG": "preprod-dev42",
  "CI_ENVIRONMENT_URL": "https://preprod-sample.dev42.fabrique.social.gouv.fr",
  "CI_PROJECT_DIR": "/builds/SocialGouv/sample",
  "CI_PROJECT_ID": "85",
  "CI_PROJECT_NAME": "sample",
  "CI_PROJECT_NAMESPACE": "SocialGouv",
  "CI_PROJECT_PATH": "SocialGouv/sample",
  "CI_PROJECT_PATH_SLUG": "socialgouv-sample",
  "CI_PROJECT_ROOT_NAMESPACE": "SocialGouv",
  "CI_PROJECT_TITLE": "sample",
  "CI_PROJECT_URL": "https://gitlab.factory.social.gouv.fr/SocialGouv/sample",
  "CI_PROJECT_VISIBILITY": "public",
  "CI_REGISTRY_IMAGE": "registry.gitlab.factory.social.gouv.fr/socialgouv/sample",
  "CI_REGISTRY_USER": "gitlab-ci-token",
  "CI_REPOSITORY_URL": "https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/sample.git",
  "KUBE_INGRESS_BASE_DOMAIN": "dev42.fabrique.social.gouv.fr",
  "KUBE_NAMESPACE": "sample-85-preprod-dev42",
  "PRODUCTION": "",
}
`;

exports[`should generate foo gitlab project env: prod 1`] = `
Object {
  "CI_COMMIT_BEFORE_SHA": "0000000000000000000000000000000000000000",
  "CI_COMMIT_DESCRIPTION": "",
  "CI_COMMIT_MESSAGE": "chore(release): version 1.2.3",
  "CI_COMMIT_REF_NAME": "v1.2.3",
  "CI_COMMIT_REF_PROTECTED": "false",
  "CI_COMMIT_REF_SLUG": "v1-2-3",
  "CI_COMMIT_SHA": "0123456789abcdefghijklmnopqrstuvwxyz0123",
  "CI_COMMIT_SHORT_SHA": "0123456",
  "CI_COMMIT_TAG": "v1.2.3",
  "CI_COMMIT_TIMESTAMP": "1111-11-11T11:11:11+00:00",
  "CI_COMMIT_TITLE": "chore(release): version 1.2.3",
  "CI_ENVIRONMENT_NAME": "prod2",
  "CI_ENVIRONMENT_SLUG": "prod2",
  "CI_ENVIRONMENT_URL": "https://sample.fabrique.social.gouv.fr",
  "CI_PROJECT_DIR": "/builds/SocialGouv/sample",
  "CI_PROJECT_ID": "85",
  "CI_PROJECT_NAME": "sample",
  "CI_PROJECT_NAMESPACE": "SocialGouv",
  "CI_PROJECT_PATH": "SocialGouv/sample",
  "CI_PROJECT_PATH_SLUG": "socialgouv-sample",
  "CI_PROJECT_ROOT_NAMESPACE": "SocialGouv",
  "CI_PROJECT_TITLE": "sample",
  "CI_PROJECT_URL": "https://gitlab.factory.social.gouv.fr/SocialGouv/sample",
  "CI_PROJECT_VISIBILITY": "public",
  "CI_REGISTRY_IMAGE": "registry.gitlab.factory.social.gouv.fr/socialgouv/sample",
  "CI_REGISTRY_USER": "gitlab-ci-token",
  "CI_REPOSITORY_URL": "https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/sample.git",
  "KUBE_INGRESS_BASE_DOMAIN": "fabrique.social.gouv.fr",
  "KUBE_NAMESPACE": "sample",
  "PRODUCTION": "true",
}
`;
